env/work/srw_python/multielectron_test.py

```diff
@@ -0,0 +1,131 @@
+    ['sm_meth', 'i', 1, 'method to use for spectrum vs photon energy calculation in case of arbitrary input magnetic field: 0- "manual", 1- "auto-undulator", 2- "auto-wiggler", -1- dont use this accurate integration method (rather use approximate if possible)'],
+    ['sm_prec', 'f', 0.01, 'relative precision for spectrum vs photon energy calculation in case of arbitrary input magnetic field (nominal value is 0.01)'],
+    ['sm_nm', 'i', 10, 'number of macro-electrons to average on each node at parallel (MPI-based) calculation of spectrum in case of arbitrary input magnetic field'],
+    ['sm_ns', 'i', 5, 'saving periodicity (in terms of macro-electrons) for intermediate intensity at calculation of multi-electron spectrum in case of arbitrary input magnetic field'],
+    ['sm_type', 'i', 1, 'calculate flux (=1) or flux per unit surface (=2)'],
+    ['sm_pol', 'i', 6, 'polarization component to extract after calculation of multi-e flux or intensity: 0- Linear Horizontal, 1- Linear Vertical, 2- Linear 45 degrees, 3- Linear 135 degrees, 4- Circular Right, 5- Circular Left, 6- Total'],
+    ['sm_rm', 'i', 1, 'method for generation of pseudo-random numbers for e-beam phase-space integration: 1- standard pseudo-random number generator, 2- Halton sequences, 3- LPtau sequences (to be implemented)'],
+    ['sm_fn', 's', 'res_spec_me.dat', 'file name for saving calculated milti-e spectrum vs photon energy'],
+    ['sm_pl', 's', '', 'plot the resulting spectrum-e spectrum in a graph: ""- dont plot, "e"- show plot vs photon energy'],
+
+    #to add options
+    ['op_r', 'f', 20.0, 'longitudinal position of the first optical element [m]'],
+
+    # Former appParam:
+    ['source_type', 's', 't', 'source type, (u) idealized undulator, (t), tabulated undulator, (m) multipole, (g) gaussian beam'],
+])
+
+def _check_finite(file_path):
+    try:
+        import numpy as np
+        NUMPY = True
+    except ImportError:
+        NUMPY = False
+
+    if NUMPY:
+        with open(file_path) as f:
+            data = np.loadtxt(f)
+        condition = np.isfinite(data).all()
+    else:
+        import math
+        import uti_io
+        data = uti_io.read_ascii_data_cols(file_path, '\t', _i_col_start=0, _i_col_end=0, _n_line_skip=10)[0]
+        first_value = data[0]
+        condition = math.isinf(first_value) and not math.isnan(first_value)
+
+    assert condition, \
+        'data is not finite: [{}, ...]'.format(', '.join([str(x) for x in data[:10]]))
+
+    return data
+
+if __name__ == '__main__':
+    import os
+    import numpy as np
+    v = srwl_bl.srwl_uti_parse_options(varParam, use_sys_argv=False)
+    source_type, mag = srwl_bl.setup_source(v)
+    op = None
+    srwl_bl.SRWLBeamline(_name=v.name, _mag_approx=mag).calc_all(v, op)
+    file_path = os.path.join(v.fdir, v.sm_fn)
+    try:
+        data = _check_finite(file_path)
+    except AssertionError:
+        exit(1)
```

Comment:
We need to talk and decide if we really need this thing. If SRW returns "Nan"s - this should be treated as a bug/issue in SRW. 
Also, if this file is really required, it should obey the SRW file naming conventions: all our *.py file names have some prefixes, that put them to some category (e.g. srwl_uti_*).

Reply:
I've modified this file name: "multielectron_test.py" -> "srwl_test_multi_e.py" in line with file naming conventions in SRW.

Reply:
After thinking and testing I decided not to include the *.sh file to the main SRW repository and I would like the related line(s) to be removed from the main Makefile (I'll do it myself). 
Reasons:
- This script is not cross-platform; it does not have any value for our native Windows versions.
- It may be difficult to support and extend it in the future. It would be much better to have it in Python.
- I compiled a good SRW version for Linux (in /home/expdev/SoftDev/SRW_Dev). It passes well all examples; but at the main compilation I still get: “srwl_test_multi_e.py: FAIL (data is not finite)”. Maybe something is wrong with my file renamings, maybe Rob did not finish something there…

The idea is good though. I will be happy to include a cross-platform Python version of this *.sh test file when it will be available. We’ll be able to make more sophisticated tests there.